packages/botonic-plugin-watson/src/index.js

```diff
@@ -1,36 +1,99 @@
-import AssistantV1 from 'ibm-watson/assistant/v1'
-import IamAuthenticator from 'ibm-watson/auth'
+import axios from 'axios'
+
+import WatsonOutputParser from './watson-output-parser'
 
 export default class BotonicPluginWatson {
+  REQUIRED_OPTION_FIELDS = ['apikey', 'url', 'assistant_id']
+
+  EXPECTED_SESSION_ID_REQUEST_STATUS = 201
+  EXPECTED_OUTPUT_REQUEST_STATUS = 200
+
   constructor(options) {
-    this.options = options
+    try {
+      this.validateOptions(options)
+      this.init(options)
+    } catch (e) {
+      console.error(String(e))
+    }
+  }
+
+  validateOptions(options) {
+    if (!options) throw new Error('No options provided.')
+    if (!this.areRequiredOptionsProvided(options))
+      throw new Error('Missing one or more fields.')
+    if (!this.areRequiredOptionsDefined(options))
+      throw new Error('One or more fields are not defined.')
+  }
+
+  areRequiredOptionsProvided(options) {
+    const requiredOptionsFieldsProvided = Object.keys(options).filter(field =>
+      this.REQUIRED_OPTION_FIELDS.includes(field)
+    )
+    return (
+      requiredOptionsFieldsProvided.length == this.REQUIRED_OPTION_FIELDS.length
+    )
+  }
+
+  areRequiredOptionsDefined(options) {
+    const requiredOptionsProvided = Object.entries(
```

Comment:
read clean code book chapter on names .
"the length of a name should be correspond to the size of its scope." https://dzone.com/articles/naming-conventions-from-uncle-bobs-clean-code-phil